File: app/src/main/kt/estyle/teabaike/adapter/MainPagerAdapter.java
package estyle.teabaike.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

import estyle.teabaike.bean.ChannelBean;
import estyle.teabaike.constant.Url;
import estyle.teabaike.fragment.MainFragment;

public class MainPagerAdapter extends FragmentPagerAdapter {

    private List<ChannelBean> mChannelList;

    public MainPagerAdapter(FragmentManager fm) {
        super(fm);
        mChannelList = new ArrayList<>();
        for (int i = 0; i < Url.TYPES.length; i++) {
            mChannelList.add(new ChannelBean(Url.TITLES[i], MainFragment.newInstance(Url.TYPES[i])));
        }
    }

    @Override
    public Fragment getItem(int position) {
        return mChannelList.get(position).getFragment();
    }

    @Override
    public int getCount() {
        return mChannelList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mChannelList.get(position).getChannel();
    }
}

File: app/src/main/kt/estyle/teabaike/adapter/MainPagerAdapter.kt
package estyle.teabaike.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import estyle.teabaike.config.Url
import estyle.teabaike.fragment.MainFragment
import java.util.*

class MainPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val fragmentList by lazy { ArrayList<MainFragment>() }

    init {
        for (i in Url.TITLES.indices) {
            fragmentList.add(MainFragment.newInstance(Url.TITLES[i], Url.TYPES[i]))
        }
    }

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragmentList[position].title
    }
}
